client/my-sites/themes/index.node.js

```diff
@@ -11,21 +20,78 @@ import {
 	redirectToThemeDetails,
 } from './controller';
 import { validateFilters, validateVertical } from './validate-filters';
+import { getLanguage, getLanguageRouteParam } from 'calypso/lib/i18n-utils';
+import { setLocaleRawData } from 'calypso/state/ui/language/actions';
+
+const translationsCache = {};
+
+export function setupLocale( context, next ) {
+	if ( ! context.params.lang ) {
+		const localeDataPlaceholder = { '': {} };
+		context.store.dispatch( setLocaleRawData( localeDataPlaceholder ) ); // Reset to default locale
+
+		next();
+		return;
+	}
+	const subdomainsToLocales = Object.entries( localesToSubdomains ).reduce(
+		( acc, [ key, value ] ) => {
+			return {
+				...acc,
+				[ value ]: key,
+			};
+		},
+		{}
+	);
+	const langSlug = subdomainsToLocales[ context.params.lang ] || context.params.lang;
+	const language = getLanguage( langSlug );
+
+	if ( language ) {
+		context.lang = language.langSlug;
+		context.isRTL = language.rtl ? true : false;
+
+		if ( typeof translationsCache[ context.lang ] === 'undefined' ) {
+			translationsCache[ context.lang ] = JSON.parse(
+				fs.readFileSync(
+					path.join(
+						__dirname,
+						'..',
+						'..',
+						'..',
+						'public',
+						'evergreen',
```

Comment:
Instead of hardcoded `evergreen`, this should be a `target` variable. Readable from `context.target`.

It would be also nice to extract this unwieldy path reference to a helper:
```js
getAssetFilePath( target, 'languages/cs-v1.1.json' )
```
It would be used at many places in the language-manifest related code.